conjure-go-client/httpclient/client.go

```diff
@@ -145,7 +144,11 @@ func (c *clientImpl) doOnce(
 	if b.method == "" {
 		return nil, false, werror.ErrorWithContextParams(ctx, "httpclient: use WithRequestMethod() to specify HTTP method")
 	}
-	reqURI := joinURIAndPath(baseURI, b.path)
+	baseURL, err := url.Parse(baseURI)
+	if err != nil {
+		return nil, false, werror.WrapWithContextParams(ctx, err, "invalid URL")
```

Comment:
`NewRequestWithContext` is also going to parse this into a *url.URL, so this is no more (or less) strict than before.